sdk/event/event.go

```diff
@@ -31,26 +31,25 @@ func New(ps *pubsub.PubSub, service *servicesdk.Service, instance *instancesdk.I
 }
 
 // Emit emits a MESG event eventKey with eventData for service token.
-func (e *Event) Emit(instanceHash hash.Hash, eventKey string, eventData map[string]interface{}) error {
+func (e *Event) Emit(instanceHash hash.Hash, eventKey string, eventData map[string]interface{}) (*event.Event, error) {
```

Comment:
```suggestion
func (e *Event) Emit(evnt *event.Event) (error) {
```
what about change the parameters of this function to directly accept the event struct?
I think it will make this function a bit simpler as well as the api package.

Reply:
It's fine like that, otherwise, we will still have the confusion between the actual inputs and the resource itself which is not good. We will have the same problem that we have with the service.create

Reply:
I'm ok with both - just another thing we can make consistent accross source code.

Reply:
The function create in serviceSDK accept directly the service:
```
func (s *Service) Create(srv *service.Service) (*service.Service, error) {
```

Reply:
Can I upadte this when I'm going to remove coreapi methods?